WK2-878 Performance guide (#75)
wealthkernel/api-documentation-guides: docs/performance/Basics.md

---
tags: [Performance]
---

# Performance basics

> There are many different methods to calculate performance. We offer an API for one such method, Time Weight Rate of Return (TWRR), as the effect of cash flow can be filtered out. We calculate two figures using this methodology, one gross figure and one net of fees. 

Reviewer: ```suggestion
WealthKernel calculates performance using the Time-Weighted Rate of Return (TWRR) methodology, as this provides a better view of performance for investors by filtering out the effect of cash flow. We calculate two figures using this methodology: a gross figure and a net-of-fees figure.
```

Performance intervals are daily, except for weekends where Saturday to Monday is treated as a single interval.

<!-- theme: warning -->
> If WealthKernel does not know how your fees are accrued, these figures will not be accurate.

# How does our performance calculation work?

For a given day we calculate the rate of return as 

<!--TODO: Could make this mermaid diagram when stoplight supports mermaid v10.9.0 https://mermaid.js.org/config/math.html -->

> ![grossperf-3.png](../../assets/images/grossperf-3.png)

Author: That was the first thing I tried, sadly not. Then I saw mermaid does, but we're on too old a version 😒

or

![netperf-3.png](../../assets/images/netperf-3.png)

> In other words, we take the value at the end of the current interval, add on expected income, subtract cashflow, (and accrued fees for the net figure) and divide by the end value at the end of the previous interval.

Reviewer: ```suggestion
We take the value at the end of the current interval, add the expected income, subtract the cashflow, for the net figure any accrued fees are also subtracted. We then divide this by the end value from the end of the previous interval.
```

<!-- theme: info -->
> ExpectedIncome here represents dividends which have been declared but not yet paid. This helps smooth out spikes which can be seen in the examples.

# Examples

Assume a £100 deposit, a £2 increase in value after holding the assets for a day, and £1 on days after. For illustrative purposes settlement is assumed to be T+1.

## Simple

In the most simple example, there are no fees, dividends or cash flow. For a value of £100, measuring the TWRR for an increase in value to £102 the next day gives a value of 0.02 or 2%.

|Day|StartValue|EndValue|Performance Gross|Performance Net|
|--------|--------|--------|--------|--------|
|Day n|£0|£100|-|-|
|Day n+1|£100|£100|((100+0-0)/100)-1=0 or 0%|((100+0-0)/100)-1=0 or 0%|

> |Day n+2|£100|£102|((102+0-0)/100)-1=0.02 or 2%|((102+0-0-0)/100)-1=0.02 or 2%|

Reviewer: We should add some spacing to the calculations to make them easier to read.
